### PR TITLE
feat: Add support for pubsub for Dead Letter queue

### PR DESCRIPTION
**Pull request by asthamohta**
Screenshot 2024-08-23 at 4 15 44 PM

### CODE REVIEW

File: common/constants/constants.go [hunk 1]
```diff
@@ -90,14 +90,17 @@ const (
...
-	PUBSUB_RESOURCE         string = "pubsub"
-	PUBSUB_TOPIC_RESOURCE   string = "pubsub_topic"
-	PUBSUB_SUB_RESOURCE     string = "pubsub_sub"
-	MONITORING_RESOURCE     string = "monitoring"
-	AGG_MONITORING_RESOURCE string = "aggregated_monitoring"
-	DATASTREAM_RESOURCE     string = "datastream"
-	GCS_RESOURCE            string = "gcs"
+	DATAFLOW_RESOURCE         string = "dataflow"
+	PUBSUB_RESOURCE           string = "pubsub"
+	DLQ_PUBSUB_RESOURCE       string = "pubsub"
```
Comment (bharadwaj-aditya) [inline common/constants/constants.go:95]: nit: let's call this dlq_pubsub to be consistent with the topic and resources

File: sources/postgres/infoschema.go [hunk 2]
```diff
@@ -75,11 +75,16 @@ func (isi InfoSchemaImpl) StartChangeDataCapture(ctx context.Context, conv *inte
 	if err != nil {
 		return nil, fmt.Errorf("error reading streaming config: %v", err)
 	}
-	pubsubCfg, err := streaming.CreatePubsubResources(ctx, isi.MigrationProjectId, streamingCfg.DatastreamCfg.DestinationConnectionConfig, isi.TargetProfile.Conn.Sp.Dbname)
+	pubsubCfg, err := streaming.CreatePubsubResources(ctx, isi.MigrationProjectId, streamingCfg.DatastreamCfg.DestinationConnectionConfig, isi.TargetProfile.Conn.Sp.Dbname, "data")
```
Comment (bharadwaj-aditya) [inline sources/postgres/infoschema.go:78]: suggest we use a constants here instead of strings

**Review by bharadwaj-aditya — COMMENTED** [review]
some minor comments.

File: streaming/streaming.go [hunk 3]
```diff
@@ -520,7 +521,10 @@ func FetchTargetBucketAndPath(ctx context.Context, datastreamClient *datastream.
 	gcsProfile := res.Profile.(*datastreampb.ConnectionProfile_GcsProfile).GcsProfile
 	bucketName := gcsProfile.Bucket
 	prefix := gcsProfile.RootPath + datastreamDestinationConnCfg.Prefix
-	prefix = utils.ConcatDirectoryPath(prefix, "data/")
+	if pubsubDestination == constants.DLQ_GCS {
```
Comment (aksharauke) [inline streaming/streaming.go:524]: A comment would help

**Review by aksharauke — APPROVED** [review]
LGTM from the backend pov. The UI part am not proficient to review it.

**Pull request merged**